verify/cert.go

```diff
@@ -1,20 +1,30 @@
 package verify
 
 import (
+	"bytes"
+	"context"
+	"errors"
 	"fmt"
+	"math/big"
 
 	binding "github.com/Layr-Labs/eigenda/contracts/bindings/EigenDAServiceManager"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
 
 	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/ethereum/go-ethereum/log"
 	"golang.org/x/exp/slices"
 )
 
+var ErrBatchMetadataHashNotFound = errors.New("BatchMetadataHash not found for BatchId")
+
 // CertVerifier verifies the DA certificate against on-chain EigenDA contracts
 // to ensure disperser returned fields haven't been tampered with
 type CertVerifier struct {
-	manager *binding.ContractEigenDAServiceManagerCaller
+	ethConfirmationDepth uint64
+	manager              *binding.ContractEigenDAServiceManagerCaller
+	finalizedBlockClient *FinalizedBlockClient
+	ethClient            *ethclient.Client
```

Comment:
Is this client used anywhere in this struct?

Reply:
On line 119 yeah

```
blockHeader, err := cv.ethClient.BlockByNumber(context.Background(), nil)
```